src/rule-parser.js

```diff
@@ -141,27 +140,16 @@ class EventRequirement extends  Requirement {
   }
 
   conditionFulfilled (condition, matchedEvent) {
-    /*switch(condition) {
-    case condition['parameter'] !== undefined:
-      console.log('BANANARAMA')
-      return condition['value'] === matchedEvent['context'][condition['parameter']];
-    case condition['AnyOf'] !== undefined:
-      return this.checkAnyOf(condition['AnyOf'], matchedEvent);
-    case condition['OneOf'] !== undefined:
-      return this.checkOneOf(condition['OneOf'], matchedEvent);
-    default:
-      throw new Error(`Invalid Condition params: ${condition}`);*/
-    if (condition['parameter'] !== undefined) {
-      console.log('BANANARAMA')
-      return condition['value'] === matchedEvent['context'][condition['parameter']];
+    switch(true) {
+      case condition['parameter'] !== undefined:
+        return condition['value'] === matchedEvent['context'][condition['parameter']];
+      case condition['AnyOf'] !== undefined:
+        return this.checkAnyOf(condition['AnyOf'], matchedEvent);
+      case condition['OneOf'] !== undefined:
+        return this.checkOneOf(condition['OneOf'], matchedEvent);
+      default:
+        throw new Error(`Invalid Condition params: ${JSON.stringify(condition)}`);
     }
-    if (condition['AnyOf'] !== undefined) {
-      return this.checkAnyOf(condition['AnyOf'], matchedEvent);
-    }
-    if (condition['OneOf'] !== undefined) {
-      return this.checkOneOf(condition['OneOf'], matchedEvent);
-    }
-    throw new Error(`Invalid Condition params: ${condition}`);
```

Comment:
I can also remove the `switch` and add the `if` statements - whatever you like better.

Reply:
No, please leave the switch, thats way prettier. We just couldn't get it to work^^